File: java/src/test/java/com/jamesansley/LengthOfLastWordTest.java
package com.jamesansley;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class LengthOfLastWordTest {
    @Test
    public void exampleCases() {
        LengthOfLastWord solution = new LengthOfLastWord();
        String case2 = "   fly me   to   the moon  ";

        assertEquals(5, solution.lengthOfLastWord("Hello World"));
        assertEquals(4, solution.lengthOfLastWord(case2));
        assertEquals(6, solution.lengthOfLastWord("luffy is still joyboy"));
    }

    @Test
    public void extraCases() {
        LengthOfLastWord solution = new LengthOfLastWord();
        assertEquals(3, solution.lengthOfLastWord("foo"));
        assertEquals(3, solution.lengthOfLastWord("bar foo  "));
        assertEquals(1, solution.lengthOfLastWord(" a"));
        assertEquals(1, solution.lengthOfLastWord(" a "));
        assertEquals(1, solution.lengthOfLastWord("a "));
    }
}

File: javascript/tests/length_of_last_word_test.js
import {assertEquals} from "@std/assert";
import {lengthOfLastWord} from "@/length_of_last_word.js";

Deno.test(function exampleCases() {
    assertEquals(lengthOfLastWord("Hello World"), 5);
    const case2 = "   fly me   to   the moon  ";
    assertEquals(lengthOfLastWord(case2), 4);
    assertEquals(lengthOfLastWord("luffy is still joyboy"), 6);
});

Deno.test(function extraCases() {
    assertEquals(lengthOfLastWord("foo"), 3);
    assertEquals(lengthOfLastWord(" a"), 1);
    assertEquals(lengthOfLastWord(" a "), 1);
    assertEquals(lengthOfLastWord("a "), 1);
});
